Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Cancel.hs

```diff
@@ -103,6 +107,18 @@ cancel bookingId _ req = do
           Right res' -> do
             let merchantOperatingCityId = booking.merchantOperatingCityId
             disToPickup <- driverDistanceToPickup booking.merchantId merchantOperatingCityId (getCoordinates res'.currPoint) (getCoordinates booking.fromLocation)
+            now <- getCurrentTime
+            let driverBookingDuration = roundToIntegral $ diffUTCTime now booking.createdAt
+            distanceRes <-
```

Comment:
If driver doesn't go in pickup direction, then too this value will be updated accordingly. Need to think on this

Reply:
Yes, direction wise